File: src/VisualStudio/CSharp/Impl/Options/AutomationObject/AutomationObject.ExtractMethod.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.

using Microsoft.CodeAnalysis.ExtractMethod;

namespace Microsoft.VisualStudio.LanguageServices.CSharp.Options
{
    public partial class AutomationObject
    {
        public int ExtractMethod_AllowBestEffort
        {
            get { return GetBooleanOption(ExtractMethodPresentationOptionsStorage.AllowBestEffort); }
            set { SetBooleanOption(ExtractMethodPresentationOptionsStorage.AllowBestEffort, value); }
        }

        public int ExtractMethod_DoNotPutOutOrRefOnStruct
        {
            get { return GetBooleanOption(ExtractMethodOptionsStorage.DoNotPutOutOrRefOnStruct); }
            set { SetBooleanOption(ExtractMethodOptionsStorage.DoNotPutOutOrRefOnStruct, value); }
        }
    }
}

File: src/VisualStudio/VisualBasic/Impl/Options/AutomationObject/AutomationObject.ExtractMethod.vb
' Licensed to the .NET Foundation under one or more agreements.
' The .NET Foundation licenses this file to you under the MIT license.
' See the LICENSE file in the project root for more information.

Imports Microsoft.CodeAnalysis.ExtractMethod

Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic.Options
    Partial Public Class AutomationObject
        Public Property ExtractMethod_AllowBestEffort As Boolean
            Get
                Return GetBooleanOption(ExtractMethodPresentationOptionsStorage.AllowBestEffort)
            End Get
            Set(value As Boolean)
                SetBooleanOption(ExtractMethodPresentationOptionsStorage.AllowBestEffort, value)
            End Set
        End Property
    End Class
End Namespace
